Source/Initialization/WarpXInitData.cpp

```diff
@@ -1228,3 +1287,174 @@ void WarpX::CheckKnownIssues()
+
+    auto axisLabels = F.getAttribute("axisLabels").get<std::vector<std::string>>();
+    auto fileGeom = F.getAttribute("geometry").get<std::string>();
+
+#if defined(WARPX_DIM_3D)
+    WARPX_ALWAYS_ASSERT_WITH_MESSAGE(fileGeom == "cartesian", "3D can only read from files with cartesian geometry");
+    WARPX_ALWAYS_ASSERT_WITH_MESSAGE(axisLabels[0] == "x" && axisLabels[1] == "y" && axisLabels[2] == "z",
+                                     "3D expects axisLabels {x, y, z}");
+#elif defined(WARPX_DIM_XZ)
+    WARPX_ALWAYS_ASSERT_WITH_MESSAGE(fileGeom == "cartesian", "XZ can only read from files with cartesian geometry");
+    WARPX_ALWAYS_ASSERT_WITH_MESSAGE(axisLabels[0] == "x" && axisLabels[1] == "z",
+                                     "XZ expects axisLabels {x, z}");
+#elif defined(WARPX_DIM_RZ)
+    WARPX_ALWAYS_ASSERT_WITH_MESSAGE(fileGeom == "thetaMode", "RZ can only read from files with 'thetaMode'  geometry");
+    WARPX_ALWAYS_ASSERT_WITH_MESSAGE(axisLabels[0] == "r" && axisLabels[1] == "z",
+                                     "RZ expects axisLabels {r, z}");
+#endif
+
+    auto offset = F.gridGlobalOffset();
+    amrex::Real offset0 = offset[0];
+    amrex::Real offset1 = offset[1];
+#if defined(WARPX_DIM_3D)
+    amrex::Real offset2 = offset[2];
+#endif
+    auto d = F.gridSpacing<long double>();
+    amrex::Real d0 = d[0];
+    amrex::Real d1 = d[1];
+#if defined(WARPX_DIM_3D)
+    amrex::Real d2 = d[2];
+#endif
+
+    auto FC = F[F_component];
+    auto extent = FC.getExtent();
+    int extent0 = extent[0];
+    int extent1 = extent[1];
+    int extent2 = extent[2];
+
+    // Determine the chunk data that will be loaded.
+    // Now, the full range of data is loaded.
+    // Loading chunk data can speed up the process.
+    // Thus, `chunk_offset` and `chunk_extent` should be modified accordingly in another PR.
+    openPMD::Offset chunk_offset = {0,0,0};
+    openPMD::Extent chunk_extent = {extent[0], extent[1], extent[2]};
+
+    auto FC_chunk_data = FC.loadChunk<double>(chunk_offset,chunk_extent);
+    series.flush();
+    auto FC_data_host = FC_chunk_data.get();
+
+    // Load data to GPU
+    size_t total_extent = size_t(extent[0]) * extent[1] * extent[2];
+    amrex::Gpu::DeviceVector<double> FC_data_gpu(total_extent);
+    auto FC_data = FC_data_gpu.data();
+    amrex::Gpu::copy(amrex::Gpu::hostToDevice, FC_data_host, FC_data_host + total_extent, FC_data);
+
+    // Loop over boxes
+    for (MFIter mfi(*mf, TilingIfNotGPU()); mfi.isValid(); ++mfi)
+    {
+        amrex::Box box = mfi.growntilebox();
+        amrex::Box tb = mfi.tilebox(nodal_flag, mf->nGrowVect());
+        auto const& mffab = mf->array(mfi);
+
+        // Start ParallelFor
+        amrex::ParallelFor (tb,
+            [=] AMREX_GPU_DEVICE (int i, int j, int k) {
+                // i,j,k denote x,y,z indices in 3D xyz.
+                // i,j,k denote r,z,mode indices in 2D rz.
```

Comment:
Also discussed with @RemiLehe offline, we are not sure that `k` here represents the mode index. Isn't this maybe just an empty index, which can get only one single value (`0`) in RZ (aka, in the 2D version of AMReX's `ParallelFor`), rather than the mode index spanning from `0` to the number of modes?

Reply:
Fixed.